crates/sui-types/src/passkey_authenticator.rs

```diff
@@ -0,0 +1,309 @@
+            return Err(SuiError::InvalidSignature {
+                error: "Invalid challenge".to_string(),
+            });
+        };
+
+        // Construct msg = authenticator_data || sha256(client_data_json).
+        let mut message = self.authenticator_data.clone();
+        let client_data_hash = Sha256::digest(self.client_data_json.as_bytes()).digest;
+        message.extend_from_slice(&client_data_hash);
+
+        // Check if author is derived from the public key.
+        if author != SuiAddress::from(&self.get_pk()?) {
+            return Err(SuiError::InvalidSignature {
+                error: "Invalid author".to_string(),
+            });
+        };
+
+        // Verify the signature against pk and message.
+        self.pk
+            .verify(&message, &self.signature)
+            .map_err(|_| SuiError::InvalidSignature {
+                error: "Fails to verify".to_string(),
+            })
+    }
+}
+
+impl ToFromBytes for PasskeyAuthenticator {
+    fn from_bytes(bytes: &[u8]) -> Result<Self, FastCryptoError> {
+        // The first byte matches the flag of PasskeyAuthenticator.
+        if bytes.first().ok_or(FastCryptoError::InvalidInput)?
+            != &SignatureScheme::PasskeyAuthenticator.flag()
+        {
+            return Err(FastCryptoError::InvalidInput);
+        }
+        let passkey: PasskeyAuthenticator =
+            bcs::from_bytes(&bytes[1..]).map_err(|_| FastCryptoError::InvalidSignature)?;
+        Ok(passkey)
+    }
+}
+
+impl AsRef<[u8]> for PasskeyAuthenticator {
+    fn as_ref(&self) -> &[u8] {
+        self.bytes
+            .get_or_try_init::<_, eyre::Report>(|| {
+                let as_bytes = bcs::to_bytes(self).expect("BCS serialization should not fail");
+                let mut bytes = Vec::with_capacity(1 + as_bytes.len());
+                bytes.push(SignatureScheme::PasskeyAuthenticator.flag());
+                bytes.extend_from_slice(as_bytes.as_slice());
+                Ok(bytes)
+            })
+            .expect("OnceCell invariant violated")
+    }
+}
+/// Compute the digest that the signature committed over as `intent || hash(tx_data)`, total
+/// of 3 + 32 = 35 bytes.
+pub fn to_signing_message<T: Serialize>(
+    intent_msg: &IntentMessage<T>,
+) -> [u8; INTENT_PREFIX_LENGTH + DefaultHash::OUTPUT_SIZE] {
+    let mut extended = [0; INTENT_PREFIX_LENGTH + DefaultHash::OUTPUT_SIZE];
+    extended[..INTENT_PREFIX_LENGTH].copy_from_slice(&intent_msg.intent.to_bytes());
+    extended[INTENT_PREFIX_LENGTH..].copy_from_slice(&to_tx_digest(intent_msg));
+    extended
+}
+
+/// Compute the hash(tx_data) from the intent msg.
+pub fn to_tx_digest<T: Serialize>(intent_msg: &IntentMessage<T>) -> [u8; DefaultHash::OUTPUT_SIZE] {
+    let mut hasher = DefaultHash::default();
+    bcs::serialize_into(&mut hasher, &intent_msg.value)
+        .expect("Message serialization should not fail");
+    hasher.finalize().digest
+}
```

Comment:
So this is actually incorrect as its not properly taking into account the salt "TransactionData::" that is required for actually computing the transaction digest. this means the output of this function and `Transaction::digest` will differ.

Reply:
good point, revised to signing digest